### PR TITLE
Add tox config and build using CI

### LINKED ISSUE #469: Add tox or other test orchestrator to run tests locally
Today, I started to run the tests on this project only to learn that there's no orchestration for running the tests. The user is required to know the magical setup to run the tests or manually replicate the [steps used by ci](https://github.com/python/bedevere/blob/ad03d53e210b9c2f84733ef8f264e6831b062e40/.github/workflows/ci.yml#L27-L47). I'd prefer if there were a one-command guide on how to run the tests for the project.

It looks like there is a one-liner that works to run the tests:

```
pip-run -q -r dev-requirements.txt -- -m pytest
```

Of course, it depends on pip-run. Most Python developers are more familiar with something like tox or nox.

Any reason this project doesn't provide such a thing? If it provides, it, I'd also recommend to run the tests using such an orchestrator (to have one primary workflow and not divergent workflows in CI).

### PR DESCRIPTION
**Pull request by jaraco**
Fixes #469.

- Add tox.ini from jaraco/skeleton
- Trim irrelevant details from tox.ini
- In CI, invoke tests with tox. Move pytest args to pytest config.

### CODE REVIEW

**Review by ezio-melotti — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,7 +36,7 @@ jobs:
           # https://github.com/aio-libs/yarl/issues/680
           YARL_NO_EXTENSIONS: 1
       - name: Run Tests
-        run: pytest --cov --cov-report=term --cov-report=xml
+        run: tox
```
Comment (hugovk) [inline .github/workflows/ci.yml:39]: And to match the `tox.ini` listing the versions to test, I usually use this to run just for the current Python version (e.g. [here](https://github.com/hugovk/pepotron/blob/e9d0511d0e87f5c0d506dbeb7614ffc9f2e06f33/.github/workflows/test.yml#L40) and [here](https://github.com/hugovk/pepotron/runs/6949910350?check_suite_focus=true#step:5:16)):

```suggestion
        run: tox -e py
```

**Review by hugovk — COMMENTED** [review]
Looks good! A few comments.

Whilst updating `ci.yml`, we can bump this too:

```diff
-uses: actions/setup-python@v3
+uses: actions/setup-python@v4
```

File: pytest.ini [hunk 2]
```diff
@@ -1,4 +1,5 @@
 [pytest]
 testpaths = tests
+addopts = --cov --cov-report=term --cov-report=xml
```
[the remark this answers is not in the document]
Comment (jaraco) [reply pytest.ini:3]: I have no opinion here, so sure.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jaraco) [reply .github/workflows/ci.yml:42]: Fixed in be31e3a.

**Comment by jaraco** [comment]
> Looks good! A few comments.
> 
> Whilst updating `ci.yml`, we can bump this too:
> 
> ```diff
> -uses: actions/setup-python@v3
> +uses: actions/setup-python@v4
> ```

Proposed in https://github.com/python/bedevere/pull/473.

**Comment by hugovk** [comment]
This was merged before the CI finished running and has broken the CI:

```
Invalid workflow file: .github/workflows/ci.yml#L41
The workflow is not valid. .github/workflows/ci.yml (Line: 41, Col: 11): Unexpected value 'TOXENV=python'
```

https://github.com/python/bedevere/actions/runs/2529668873

**Comment by hugovk** [comment]
Please see PR https://github.com/python/bedevere/pull/474.

**Pull request merged**